setup.py

```diff
@@ -191,7 +191,14 @@ def get_executable_name(name: str) -> str:
 class _BaseExtension(Extension):
     """A base class that maps an abstract source to an abstract destination."""
 
-    def __init__(self, src: str, dst: str, name: str):
+    def __init__(
+        self,
+        src: str,
+        dst: str,
+        name: str,
+        is_cmake_built: bool = True,
+        is_dst_dir: bool = False,
```

Comment:
`dst` already has semantics for whether it's a directory or not: does it end in `/`. It is subtle, so maybe a bool is better, but if we're now using the bool then the comment on `self.dst: str = dst` needs to change.

If you do keep this bool please call it `dst_is_dir` since it describes the `dst` param and not the extension object. `extension.is_dst_dir` makes it sound like it describes the extension.

Reply:
Yeah I don't want to rely on `/` alone since the input string might be a directory but without `/`.